### ISSUE TITLE
add version support table

### ISSUE THREAD

**Issue by akaszynski** [pyansys/dev-guide#24]
Should be a generic support table that can be used by multiple repositories. This should be a best practice under the guidelines section.

Should be `rst` format so it can be a "write once use everywhere".

**Comment by akaszynski** [comment]
Version support table should follow `numpy` deprecation schedule [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html). Adding note for @koubaa as we spoke about this yesterday.

**Comment by greschd** [comment]
Maybe the lowest supported Python should be `lower(nep29_python, ansys_python)`, where `ansys_python` is the latest version included in a released installer?
Note e.g. that Python 3.7 is dropped according to NEP 29, whereas 2022R1 still ships it. If we also drop 3.7 support, we'll make it unnecessarily hard to install PyAnsys libraries into Ansys-Python environments.

**Comment by koubaa** [comment]
@akaszynski I agree with the goal and its treatment as an issue.  @greschd brings up great points.

Some options that I can spitball:
- unilaterally fix the issue here and let the Ansys product environments QA against a version that we don't support here
- push the issue downstream to those products and when its fixed there we come back and fix it here.
- not have an opinion on the downstream products and use the lesser of the two to determine version support here.
- have two levels of support, "official" support following nep29 and "unofficial" support matching minimum product python versions.  We still CI for unofficial but we let the tests fail so at least we have an account of what problems if any exist for older environments.
